### PR TITLE
fix(kafka producer): kick off recovery from disk for fixed topics, migrate old replayq directories

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes https://emqx.atlassian.net/browse/EMQX-13242
Fixes https://emqx.atlassian.net/browse/EMQX-13248

Release version: e5.8.2

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_producer.erl [hunk 1]
```diff
@@ -811,6 +824,46 @@ replayq_dir(BridgeType, BridgeName) ->
...
+            fun(File) ->
+                PartitionDir = filename:join([OldWolffDir, File]),
+                IsDir = filelib:is_dir(PartitionDir),
+                case IsDir andalso string:to_integer(File) of
+                    {_Int, Rest} when Rest =:= ""; Rest =:= <<>> ->
+                        {true, PartitionDir};
+                    _ ->
+                        false
+                end
+            end,
```
Comment (zmstone) [inline apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_producer.erl:849]: maybe impl a function like:
is_old_disk_buffer_dir(File) ->

File: apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_producer.erl [hunk 2]
```diff
@@ -159,11 +163,20 @@ create_producers_for_bridge_v2(
...
+    WolffProducerConfig =
+        #{replayq_dir := ReplayqDir} = producers_config(
+            BridgeType, BridgeName, KafkaConfig, IsDryRun, ActionResId
+        ),
+    maybe_migrate_old_replayq_dir(ReplayqDir, ActionResId, TopicType, MKafkaTopic),
     case wolff:ensure_supervised_dynamic_producers(ClientId, WolffProducerConfig) of
         {ok, Producers} ->
+            case TopicType of
+                fixed ->
+                    ok = wolff:add_topic(Producers, MKafkaTopic),
```
Comment (zmstone) [inline apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_producer.erl:175]: do not crash on `badmatch` with ok ?
maybe log and throw error instead, like line 211

Comment (thalesmg) [reply apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_producer.erl]: :checkered_flag:

File: apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_producer.erl (hunk 1 above)
Comment (thalesmg) [reply apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_producer.erl:877]: :checkered_flag:

Comment (zmstone) [reply apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_producer.erl:877]: nvm

**Review by zmstone — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-4597)

**Pull request merged**